From 77a3ffaa3a6c540dd9a89c2b2d9b4947fb70647d Mon Sep 17 00:00:00 2001
From: Takeshi Kihara <takeshi.kihara.df@renesas.com>
Date: Mon, 1 Apr 2019 19:45:34 +0900
Subject: [PATCH 793/909] Revert "arm64: dts: r8a7795: Connect SDHI to
 IPMMU-DS1"

commit b1aaf0d1ee4f944b58c30191aa8f6b953b8bddda from
git://git.kernel.org/pub/scm/linux/kernel/git/horms/renesas-bsp.git

This reverts commit 4bd36fe168284318c1091e998e7aa8351ff34528.

Multiple IPMMUs can not be operated simultaneously due to H3 Ver.1.x and
H3 Ver.2.0 hardware restriction.

As a result, this reverts commit 4bd36fe16828 ("arm64: dts: r8a7795:
Connect SDHI to IPMMU-DS1") to keep IPMMU for SDHI{0..3} is disabled by
default.

Signed-off-by: Takeshi Kihara <takeshi.kihara.df@renesas.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/renesas/r8a7795-es1.dtsi | 16 ----------------
 arch/arm64/boot/dts/renesas/r8a7795.dtsi     |  4 ----
 2 files changed, 20 deletions(-)

diff --git a/arch/arm64/boot/dts/renesas/r8a7795-es1.dtsi b/arch/arm64/boot/dts/renesas/r8a7795-es1.dtsi
index 9c160cd53cc2..81e762541f57 100644
--- a/arch/arm64/boot/dts/renesas/r8a7795-es1.dtsi
+++ b/arch/arm64/boot/dts/renesas/r8a7795-es1.dtsi
@@ -281,22 +281,6 @@
 	renesas,ipmmu-main = <&ipmmu_mm 7>;
 };
 
-&sdhi0 {
-	/delete-property/ iommus;
-};
-
-&sdhi1 {
-	/delete-property/ iommus;
-};
-
-&sdhi2 {
-	/delete-property/ iommus;
-};
-
-&sdhi3 {
-	/delete-property/ iommus;
-};
-
 &du {
 	vsps = <&vspd0 &vspd1 &vspd2 &vspd3>;
 };
diff --git a/arch/arm64/boot/dts/renesas/r8a7795.dtsi b/arch/arm64/boot/dts/renesas/r8a7795.dtsi
index cce15399e60e..2cf1ae688ce1 100644
--- a/arch/arm64/boot/dts/renesas/r8a7795.dtsi
+++ b/arch/arm64/boot/dts/renesas/r8a7795.dtsi
@@ -2804,7 +2804,6 @@
 			max-frequency = <200000000>;
 			power-domains = <&sysc R8A7795_PD_ALWAYS_ON>;
 			resets = <&cpg 314>;
-			iommus = <&ipmmu_ds1 32>;
 			status = "disabled";
 		};
 
@@ -2817,7 +2816,6 @@
 			max-frequency = <200000000>;
 			power-domains = <&sysc R8A7795_PD_ALWAYS_ON>;
 			resets = <&cpg 313>;
-			iommus = <&ipmmu_ds1 33>;
 			status = "disabled";
 		};
 
@@ -2830,7 +2828,6 @@
 			max-frequency = <200000000>;
 			power-domains = <&sysc R8A7795_PD_ALWAYS_ON>;
 			resets = <&cpg 312>;
-			iommus = <&ipmmu_ds1 34>;
 			status = "disabled";
 		};
 
@@ -2843,7 +2840,6 @@
 			max-frequency = <200000000>;
 			power-domains = <&sysc R8A7795_PD_ALWAYS_ON>;
 			resets = <&cpg 311>;
-			iommus = <&ipmmu_ds1 35>;
 			status = "disabled";
 		};
 
-- 
2.17.1

